### ISSUE TITLE
Few JS warnings

### ISSUE THREAD

**Issue by bil-elmoussaoui** [jderose9/dash-to-panel#239]
```
sep 04 04:55:59 lenovo gnome-shell[19623]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 04:56:24 lenovo gnome-shell[19623]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 04:56:24 lenovo gnome-shell[19623]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 04:56:27 lenovo gnome-shell[19623]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 04:56:55 lenovo gnome-shell[19623]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 05:05:07 lenovo gnome-shell[22913]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 05:05:17 lenovo gnome-shell[22913]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 05:05:17 lenovo gnome-shell[22913]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 05:05:17 lenovo gnome-shell[22913]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 05:05:24 lenovo gnome-shell[22913]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 05:08:03 lenovo gnome-shell[24807]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 05:08:12 lenovo gnome-shell[24807]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 05:08:12 lenovo gnome-shell[24807]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 05:08:12 lenovo gnome-shell[24807]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 05:08:24 lenovo gnome-shell[24807]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 18:15:49 lenovo gnome-shell[10340]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 18:16:01 lenovo gnome-shell[10340]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 18:16:01 lenovo gnome-shell[10340]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 18:16:01 lenovo gnome-shell[10340]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 18:16:05 lenovo gnome-shell[10340]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 18:27:50 lenovo gnome-shell[14342]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 18:27:56 lenovo gnome-shell[14342]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 18:27:56 lenovo gnome-shell[14342]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 18:27:56 lenovo gnome-shell[14342]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 18:28:11 lenovo gnome-shell[14342]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 18:29:37 lenovo gnome-shell[14883]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 18:29:50 lenovo gnome-shell[14883]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 18:29:50 lenovo gnome-shell[14883]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 18:29:50 lenovo gnome-shell[14883]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 18:29:59 lenovo gnome-shell[14883]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 19:00:13 lenovo gnome-shell[21770]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 19:00:21 lenovo gnome-shell[21770]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 19:00:21 lenovo gnome-shell[21770]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 19:00:21 lenovo gnome-shell[21770]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 19:00:34 lenovo gnome-shell[21770]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 19:07:04 lenovo gnome-shell[23658]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 19:07:14 lenovo gnome-shell[23658]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 19:07:14 lenovo gnome-shell[23658]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 19:07:15 lenovo gnome-shell[23658]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 19:07:26 lenovo gnome-shell[23658]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 19:11:19 lenovo gnome-shell[24813]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 19:11:25 lenovo gnome-shell[24813]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 19:11:25 lenovo gnome-shell[24813]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 19:11:25 lenovo gnome-shell[24813]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 19:11:40 lenovo gnome-shell[24813]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 20:01:14 lenovo gnome-shell[1188]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 20:01:20 lenovo gnome-shell[1188]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 20:01:20 lenovo gnome-shell[1188]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 20:01:21 lenovo gnome-shell[1188]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 20:01:36 lenovo gnome-shell[1188]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 20:05:32 lenovo gnome-shell[2167]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 04 20:05:38 lenovo gnome-shell[2167]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 04 20:05:38 lenovo gnome-shell[2167]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 04 20:05:38 lenovo gnome-shell[2167]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 04 20:05:54 lenovo gnome-shell[2167]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 04 20:38:24 lenovo gnome-shell[2167]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 07 05:05:23 lenovo gnome-shell[1624]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 07 05:05:32 lenovo gnome-shell[1624]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 07 05:05:32 lenovo gnome-shell[1624]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 07 05:05:32 lenovo gnome-shell[1624]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 07 05:06:22 lenovo gnome-shell[1624]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 07 05:32:59 lenovo gnome-shell[8364]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 07 05:33:04 lenovo gnome-shell[8364]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 07 05:33:04 lenovo gnome-shell[8364]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 07 05:33:05 lenovo gnome-shell[8364]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 07 05:33:45 lenovo gnome-shell[8364]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]
sep 07 23:50:13 lenovo gnome-shell[24039]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js 90]: Too many arguments to method St.Bin.get_child: expected 0, got 1
sep 07 23:50:19 lenovo gnome-shell[24039]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/taskbar.js 747]: reference to undefined property oldApps[oldIndex]
sep 07 23:50:19 lenovo gnome-shell[24039]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 786]: reference to undefined property this._stateChangedId
sep 07 23:50:20 lenovo gnome-shell[24039]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/appIcons.js 419]: reference to undefined property dots._tweeningToWidth
sep 07 23:50:55 lenovo gnome-shell[24039]: JS WARNING: [/home/bilal/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 934]: reference to undefined property oldWin[oldIndex]


```

**Comment by franglais125** [comment]
@bil-elmoussaoui is this on Gnome 3.26? Perhaps due to a newer GJS?

**Comment by bil-elmoussaoui** [comment]
Yeah it's `GNOME Shell 3.25.91`

**Comment by vans163** [comment]
JS WARNING: [/home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/windowPreview.js 963]: reference to undefined property oldWin[oldIndex]

same warning on 3.26

**Comment by jderose9** [comment]
I see these on 3.26 too. The extension seems to work fine still. Does it for you? (I think the warnings should get cleaned up regardless but I am trying to determine urgency.)

**Comment by vans163** [comment]
@jderose9 The concern is, if its passing invalid params, the extension may appear to work, but glib is so fragile, a warning/error in a plugin could easily lead to a crash in glib.  And a crash in glib will kill all your open windows on Wayland, or mess up the GUI on Xorg.

**Comment by jderose9** [comment]
I looked into this a little more. The variables are supposed to be undefined, and the behavior is the same as it was prior to 3.26. However, there seems to be a change in Gjs or Spidermonkey causing these "strict mode" warnings to be logged when a variable is used in a comparison without being assigned first. I'll put some code in that will silence them, but I don't think there is any harm from them and there are quite a few coming from gnome shell too.

**Comment by jderose9** [comment]
These should be resolved by #259 .
